bitnami/argo-cd/templates/repo-server/deployment.yaml

```
@@ -273,9 +309,82 @@ spec:
+            - name: cmp-server-plugins
+              mountPath: /home/argocd/cmp-server/plugins
+            {{- end }}
           {{- if .Values.repoServer.extraVolumeMounts }}
           {{- include "common.tplvalues.render" (dict "value" .Values.repoServer.extraVolumeMounts "context" $) | nindent 12 }}
           {{- end }}
+        {{- if .Values.repoServer.configManagementPlugins.enabled }}
+        {{- range $plugin := .Values.repoServer.configManagementPlugins.plugins }}
+        {{- $additionalBinariesDir := $plugin.additionalBinariesDir | default "/usr/local/bin" | trimSuffix "/" }}
+        - name: {{ $plugin.name }}
+          image: {{ include "common.images.image" (dict "imageRoot" $plugin.sidecar.image "global" $.Values.global) }}
+          imagePullPolicy: {{ $plugin.sidecar.image.pullPolicy | default $.Values.image.pullPolicy }}
+          {{- if $plugin.sidecar.lifecycleHooks }}
+          lifecycle: {{- include "common.tplvalues.render" (dict "value" $plugin.sidecar.lifecycleHooks "context" $) | nindent 12 }}
+          {{- end }}
+          {{- $cmpSidecarSecurityContext := include "argocd.tplvalues.merge-with-precedence" (dict "values" (list $.Values.repoServer.containerSecurityContext $plugin.sidecar.containerSecurityContext) "context" $) | fromYaml }}
+          {{- if $cmpSidecarSecurityContext.enabled }}
+          securityContext: {{- include "common.compatibility.renderSecurityContext" (dict "secContext" $cmpSidecarSecurityContext "context" $) | nindent 12 }}
+          {{- end }}
+          {{- if $plugin.sidecar.customCommand }}
+          command: {{- include "common.tplvalues.render" (dict "value" $plugin.sidecar.customCommand "context" $) | nindent 12 }}
+          {{- else }}
+          command:
+            - {{ $additionalBinariesDir }}/argocd-cmp-server
+          {{- end }}
+          {{- if $plugin.sidecar.customArgs }}
+          args: {{- include "common.tplvalues.render" (dict "value" $plugin.sidecar.customArgs "context" $) | nindent 12 }}
+          {{- end }}
+          env:
+            {{- if $plugin.sidecar.extraEnvVars }}
+            {{- include "common.tplvalues.render" (dict "value" $plugin.sidecar.extraEnvVars "context" $) | nindent 12 }}
+            {{- end }}
+          envFrom:
+            {{- if $plugin.sidecar.extraEnvVarsCM }}
+            - configMapRef:
+                name: {{ include "common.tplvalues.render" (dict "value" $plugin.sidecar.extraEnvVarsCM "context" $) }}
+            {{- end }}
+            {{- if $plugin.sidecar.extraEnvVarsSecret }}
+            - secretRef:
+                name: {{ include "common.tplvalues.render" (dict "value" $plugin.sidecar.extraEnvVarsSecret "context" $) }}
+            {{- end }}
+          {{- if $plugin.sidecar.resources }}
+          resources: {{- toYaml $plugin.sidecar.resources | nindent 12 }}
+          {{- else if and $plugin.sidecar.resourcePreset (ne $plugin.sidecar.resourcesPreset "none") }}
```

There is a typo here, but in any case, refer to my previous comment about simplifying all this code and allow users to set their sidecars config explicitly.

```suggestion
          {{- else if and $plugin.sidecar.resourcesPreset (ne $plugin.sidecar.resourcesPreset "none") }}
```